packages/teams-js/src/public/visualMedia.ts

```diff
@@ -0,0 +1,308 @@
+import { sendAndHandleSdkError } from '../internal/communication';
+import { ensureInitialized } from '../internal/internalAPIs';
+import { errorNotSupportedOnPlatform, FrameContexts } from './constants';
+import { DevicePermission } from './interfaces';
+import { runtime } from './runtime';
+
+/**
+ * Interact with image and video. It lets the app developer ask the user to get images or video from their camera / camera roll / file system.
+ * @beta
+ */
+export namespace visualMedia {
+  /**
+   * @hidden
+   * Hide from docs
+   * --------
+   * All properties common to Image and Video Props
+   * @beta
+   */
+  interface VisualMediaProps {
+    /**
+     * max limit of media allowed to be selected in one go, current max limit is 10.
+     */
+    maxVisualMediaCount: number;
+  }
+
+  /**
+   * The required value of the visualMedia files from gallery
+   * @beta
+   */
+  export interface GalleryProps {
```

Comment:
I would prefer we not continue to make `interfaces.ts` even larger. Files like "interfaces" and "constants" and "utils" have inherent scale issues because they infinitely grow and have no functionality-based associations (it's just a junk drawer for all utils or all interfaces or all constants).

I'd prefer we leave interfaces more closely associated with the functionality to which they are affinitized. It doesn't necessarily need to be in the same file (although it could be) -- but putting all interfaces in a single file just creates an ever-growing collection.

Reply:
That makes sense. I will keep the constants and interface in the current file.

Reply:
> I would prefer we not continue to make `interfaces.ts` even larger. Files like "interfaces" and "constants" and "utils" have inherent scale issues because they infinitely grow and have no functionality-based associations (it's just a junk drawer for all utils or all interfaces or all constants).
> 
> I'd prefer we leave interfaces more closely associated with the functionality to which they are affinitized. It doesn't necessarily need to be in the same file (although it could be) -- but putting all interfaces in a single file just creates an ever-growing collection.

Can we have interfaces folder in the repo where each capability will get the interface and it'll be easier to maintain. I prefer this because we can have more readable code and we can maintain it in the future. Having interfaces and constants in one file makes it harder to read and understand the code.

Reply:
How about we create an improvement task to have an interfaces folder in the repo where each capability gets its own interface? This will make it easier to maintain. 
Also, I prefer not to introduce too many topics into a single API implementation PR for two main reasons:
1. To meet the feature's deadline, this PR is blocking all subsequent check-ins.
2. I've noticed that not all APIs in the current codebase place their interfaces in `interfaces.ts`. A focused refactor makes more sense.

Reply:
> > I would prefer we not continue to make `interfaces.ts` even larger. Files like "interfaces" and "constants" and "utils" have inherent scale issues because they infinitely grow and have no functionality-based associations (it's just a junk drawer for all utils or all interfaces or all constants).
> > I'd prefer we leave interfaces more closely associated with the functionality to which they are affinitized. It doesn't necessarily need to be in the same file (although it could be) -- but putting all interfaces in a single file just creates an ever-growing collection.
> 
> Can we have interfaces folder in the repo where each capability will get the interface and it'll be easier to maintain. I prefer this because we can have more readable code and we can maintain it in the future. Having interfaces and constants in one file makes it harder to read and understand the code.

How about we create an improvement task to have an interfaces folder in the repo where each capability gets its own interface? This will make it easier to maintain.
Also, I prefer not to introduce too many topics into a single API implementation PR for two main reasons:

To meet the feature's deadline, this PR is blocking all subsequent check-ins.
I've noticed that not all APIs in the current codebase place their interfaces in interfaces.ts. A focused refactor makes more sense.

Reply:
I agree that we don't have to design the best solution the context of this PR.  😜

Reply:
Sure, I agree that we don't need to include in this PR but this is something we should fix this before the situation goes out of control down the lane. @TrevorJoelHarris Shall we create a task to refactor 
 `interfaces`? I can create and work on them side-by-side.